rotkehlchen/api/v1/schemas.py

```diff
@@ -1108,15 +1119,79 @@ def make_history_base_entry(self, data: dict[str, Any], **_kwargs: Any) -> dict[
                 receive_identifier=context_schema.get_grouped_event_identifier(
                     data=data,
                     subtype=HistoryEventSubType.RECEIVE,
+                    sequence_index_offset=1,
                 ),
                 fee_identifier=context_schema.get_grouped_event_identifier(
                     data=data,
                     subtype=HistoryEventSubType.FEE,
+                    sequence_index_offset=2,
                 ),
                 extra_data=extra_data,
             )
             return {'events': events}
 
+    class CreateEvmSwapEventSchema(BaseSwapEventSchema, BaseEvmEventSchema):
+        sequence_index = fields.Integer(required=True)
+
+        @post_load
+        def make_history_base_entry(self, data: dict[str, Any], **_kwargs: Any) -> dict[str, Any]:
+            spend_notes, receive_notes, fee_notes = self._validate_fee_and_get_notes(data)
+            context_schema = CreateHistoryEventSchema.history_event_context.get()['schema']
+            events = [EvmSwapEvent(
+                tx_hash=(tx_hash := data['tx_hash']),
+                sequence_index=(sequence_index := data['sequence_index']),
+                timestamp=(timestamp := data['timestamp']),
+                location=(location := data['location']),
+                event_subtype=HistoryEventSubType.SPEND,
+                asset=data['spend_asset'],
+                amount=data['spend_amount'],
+                location_label=data['location_label'],
+                notes=spend_notes,
+                identifier=data.get('identifier'),
+                event_identifier=(event_identifier := data['event_identifier']),
+                extra_data=data['extra_data'],
+                counterparty=data['counterparty'],
+                product=data['product'],
+                address=data['address'],
+            ), EvmSwapEvent(
+                tx_hash=tx_hash,
+                sequence_index=sequence_index + 1,
+                timestamp=timestamp,
+                location=location,
+                event_subtype=HistoryEventSubType.RECEIVE,
+                asset=data['receive_asset'],
+                amount=data['receive_amount'],
+                notes=receive_notes,
+                identifier=context_schema.get_grouped_event_identifier(
+                    data=data,
+                    subtype=HistoryEventSubType.RECEIVE,
+                    sequence_index_offset=1,
+                ),
+                event_identifier=event_identifier,
+            )]
+            if (
+                (fee_asset := data['fee_asset']) is not None and
+                (fee_amount := data['fee_amount']) != ZERO
+            ):
```

Comment:
this is already ensured by the base schema validation right? You only need to check one unless I'm mistaken